src/xunit.performance.api/ETWProfiler.cs

```diff
@@ -107,9 +116,142 @@ public static void Profile(string assemblyFileName, string sessionName, Action a
 
             TraceEventSession.MergeInPlace(userFullFileName, Console.Out);
             WriteInfoLine($"ETW Tracing Session saved to \"{userFullFileName}\"");
+            collectOutputFilesCallback(userFullFileName);
+
+            var assemblyModel = GetAssemblyModel(assemblyFileName, userFullFileName, sessionName, performanceTestMessages);
+            var fileNameWithoutExtension = Path.GetFileNameWithoutExtension(userFullFileName);
+            var xmlFileName = Path.Combine(outputDirectory, $"{fileNameWithoutExtension}.xml");
+            new AssemblyModelCollection { assemblyModel }.Serialize(xmlFileName);
+            WriteInfoLine($"Performance results saved to \"{xmlFileName}\"");
+            collectOutputFilesCallback(xmlFileName);
+
+            var mdFileName = Path.Combine(outputDirectory, $"{fileNameWithoutExtension}.md");
+            var dt = CreateStatistics(assemblyModel);
+            var mdTable = MarkdownHelper.GenerateMarkdownTable(dt);
+            MarkdownHelper.Write(mdFileName, mdTable);
+            WriteInfoLine($"Markdown file saved to \"{mdFileName}\"");
+            collectOutputFilesCallback(mdFileName);
+            Console.WriteLine(mdTable);
+
+            var csvFileName = Path.Combine(outputDirectory, $"{fileNameWithoutExtension}.csv");
+            dt.WriteToCSV(csvFileName);
+            WriteInfoLine($"Statistics written to \"{csvFileName}\"");
+            collectOutputFilesCallback(csvFileName);
+        }
+
+        private static DataTable CreateStatistics(AssemblyModel assemblyModel)
```

Comment:
I am moving it to a different place on the next PR. It cleans up some of the File IO generated by the API and duplicated code

---
In reply to: [102348467](https://github.com/Microsoft/xunit-performance/pull/191#discussion_r102348467) [](ancestors = 102348467)